test/integration/scheduler/taint_test.go

```diff
@@ -786,18 +763,55 @@ func TestTaintBasedEvictions(t *testing.T) {
 				}
 			}
 
+			// Regularly send heartbeat event to APIServer so that the cluster doesn't enter fullyDisruption mode.
+			// TODO(Huang-Wei): use "NodeDisruptionExclusion" feature to simply the below logic when it's beta.
+			heartbeatChan := make(chan struct{})
 			for i := 0; i < nodeCount; i++ {
-				// Stop the neededNode's heartbeat goroutine.
-				if neededNode.Name == fmt.Sprintf("node-%d", i) {
-					heartbeatChans[i] <- struct{}{}
-					break
+				var conditions []v1.NodeCondition
+				// If current node is not <neededNode>
+				if neededNode.Name != nodes[i].Name {
+					conditions = []v1.NodeCondition{
+						{
+							Type:   v1.NodeReady,
+							Status: v1.ConditionTrue,
+						},
+					}
+				} else {
+					c, err := nodeReadyStatus(test.nodeConditions)
+					if err != nil {
+						t.Error(err)
+					}
+					// Need to distinguish NodeReady/False and NodeReady/Unknown.
+					// If we try to update the node with condition NotReady/False, i.e. expect a NotReady:NoExecute taint
+					// we need to keep sending the update event to keep it alive, rather than just sending once.
+					if c == v1.ConditionFalse {
+						conditions = test.nodeConditions
+					} else if c == v1.ConditionUnknown {
+						// If it's expected to update the node with condition NotReady/Unknown,
+						// i.e. expect a Unreachable:NoExecute taint,
+						// we need to only send the update event once to simulate the network unreachable scenario.
+						nodeCopy := nodeCopyWithConditions(nodes[i], test.nodeConditions)
```

Comment:
what happens if you keep updating the status with NotReady/Unknown? if the status sticks, then we don't need to special case it here, which should simplify the code alot.

Reply:
Technically, we can, but I'm not sure I like it, even I was thinking about not sending the Unknown event at all. It's because we started the node lifecycle manager, so we should let it handle the "network lost" case totally.